hmp/estimators/utils.py

```diff
@@ -0,0 +1,211 @@
+
+    expected_channel_shape = (n_events, n_dims)
+    if channel_pars.shape[-2:] != expected_channel_shape:
+        raise ValueError(f"channel_pars shape {channel_pars.shape} incompatible with "
+                        f"expected shape ending in {expected_channel_shape}")
+
+    # Check time parameters
+    if time_pars.ndim < 2:
+        raise ValueError(f"time_pars must be at least 2D, got {time_pars.ndim}D")
+
+    expected_time_shape = (n_stages, 2)  # shape and scale parameters
+    if time_pars.shape[-2:] != expected_time_shape:
+        raise ValueError(f"time_pars shape {time_pars.shape} incompatible with "
+                        f"expected shape ending in {expected_time_shape}")
+
+    # Check for non-negative time parameters (scale parameters must be positive)
+    if np.any(time_pars[..., 1] <= 0):
+        raise ValueError("Scale parameters (time_pars[..., 1]) must be positive")
+
+    if np.any(time_pars[..., 0] <= 0):
+        raise ValueError("Shape parameters (time_pars[..., 0]) must be positive")
+
+
+def initialize_parameters(n_events: int, n_dims: int, method: str = 'random',
+                         distribution=None, max_scale: float = None,
+                         random_seed: Optional[int] = None) -> tuple[np.ndarray, np.ndarray]:
+    """Initialize channel and time parameters.
+
+    Parameters
+    ----------
+    n_events : int
+        Number of events
+    n_dims : int
+        Number of dimensions/channels
+    method : str
+        Initialization method ('random', 'uniform', 'zeros')
+    distribution : object, optional
+        Distribution object with shape and mean_to_scale methods
+    max_scale : float, optional
+        Maximum scale for time parameters
+    random_seed : int, optional
+        Random seed for reproducibility
+
+    Returns
+    -------
+    tuple[np.ndarray, np.ndarray]
+        Initialized (channel_pars, time_pars)
+    """
+    if random_seed is not None:
+        np.random.seed(random_seed)
+
+    # Initialize channel parameters
+    if method == 'random':
+        channel_pars = np.random.randn(n_events, n_dims) * 0.1
+    elif method == 'uniform':
+        channel_pars = np.ones((n_events, n_dims)) * 0.1
+    elif method == 'zeros':
+        channel_pars = np.zeros((n_events, n_dims))
+    else:
+        raise ValueError(f"Unknown initialization method: {method}")
+
+    # Initialize time parameters
+    n_stages = n_events + 1
+    time_pars = np.zeros((n_stages, 2))
+
+    if distribution is not None and max_scale is not None:
+        # Use distribution-specific initialization
+        time_pars[:, 0] = distribution.shape  # shape parameters
+        stage_durations = np.random.uniform(0, max_scale / n_stages, n_stages)
+        time_pars[:, 1] = [distribution.mean_to_scale(d) for d in stage_durations]
+    else:
+        # Default initialization
+        time_pars[:, 0] = 2.0  # Default shape
+        time_pars[:, 1] = 1.0  # Default scale
```

Comment:
- this is not possible right? Do you do that for an hypothetical distribution-free method? Or is this something that can already be run?